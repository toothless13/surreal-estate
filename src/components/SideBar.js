import React from "react";
import { Link } from "react-router-dom";
import "../styles/sidebar.css";

const SideBar = () => {
  return (
    <div className="sidebar">
      <h3 className="sidebar-heading">Filter By City</h3>
      <ul className="sidebar-links">
        <li className="sidebar-links-item">
          <Link to={`/?query={"city": "Manchester"}`}>Manchester</Link>
        </li>
        <li className="sidebar-links-item">
          <Link to={`/?query={"city": "Leeds"}`}>Leeds</Link>
        </li>
        <li className="sidebar-links-item">
          <Link to={`/?query={"city": "Oxford"}`}>Oxford</Link>
        </li>
        <li className="sidebar-links-item">
          <Link to={`/?query={"city": "Birmingham"}`}>Birmingham</Link>
        </li>
        <li className="sidebar-links-item clear-filter">
          <Link to="/">Clear Filter</Link>
        </li>
      </ul>
    </div>
  );
};

export default SideBar;
