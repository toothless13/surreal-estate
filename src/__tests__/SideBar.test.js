import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SideBar from "../components/SideBar";

describe("SideBar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SideBar />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/Filter By City/i)).toBeInTheDocument();
    expect(screen.getByText(/Manchester/i)).toBeInTheDocument();
    expect(screen.getByText(/Leeds/i)).toBeInTheDocument();
    expect(screen.getByText(/Oxford/i)).toBeInTheDocument();
    expect(screen.getByText(/Birmingham/i)).toBeInTheDocument();
    expect(screen.getByText(/Clear Filter/i)).toBeInTheDocument();
    expect(screen.getByText(/Price Ascending/i)).toBeInTheDocument();
    expect(screen.getByText(/Price Descending/i)).toBeInTheDocument();
    expect(screen.getByText(/Clear Sort/i)).toBeInTheDocument();
  });
});
