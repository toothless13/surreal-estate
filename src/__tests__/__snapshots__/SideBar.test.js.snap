// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sidebar"
  >
    <form
      class="sidebar-search"
    >
      <input
        type="text"
      />
      <button
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-magnifying-glass "
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
            fill="currentColor"
          />
        </svg>
      </button>
    </form>
    <h3
      class="sidebar-heading"
    >
      Filter By City
    </h3>
    <ul
      class="sidebar-links"
    >
      <li
        class="sidebar-links-item"
      >
        <a
          href="/?query={\\"city\\":\\"Manchester\\"}"
        >
          Manchester
        </a>
      </li>
      <li
        class="sidebar-links-item"
      >
        <a
          href="/?query={\\"city\\":\\"Leeds\\"}"
        >
          Leeds
        </a>
      </li>
      <li
        class="sidebar-links-item"
      >
        <a
          href="/?query={\\"city\\":\\"Oxford\\"}"
        >
          Oxford
        </a>
      </li>
      <li
        class="sidebar-links-item"
      >
        <a
          href="/?query={\\"city\\":\\"Birmingham\\"}"
        >
          Birmingham
        </a>
      </li>
      <li
        class="sidebar-links-item clear-filter"
      >
        <a
          href="/"
        >
          Clear Filter
        </a>
      </li>
    </ul>
    <h3
      class="sidebar-heading"
    >
      Sort By
    </h3>
    <ul
      class="sidebar-links"
    >
      <li
        class="sidebar-links-item"
      >
        <a
          href="/?sort={\\"price\\":1}"
        >
          Price Ascending
        </a>
      </li>
      <li
        class="sidebar-links-item"
      >
        <a
          href="/?sort={\\"price\\":-1}"
        >
          Price Descending
        </a>
      </li>
      <li
        class="sidebar-links-item clear-filter"
      >
        <a
          href="/"
        >
          Clear Sort
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
